src/NodaTime/Calendars/RegularWeekYearRule.cs

```diff
@@ -124,7 +175,19 @@ public LocalDate GetLocalDate(int weekYear, int weekOfWeekYear, IsoDayOfWeek day
                     throw new ArgumentOutOfRangeException(nameof(weekYear),
                         $"The combination of {nameof(weekYear)}, {nameof(weekOfWeekYear)} and {nameof(dayOfWeek)} is invalid");
                 }
-                return new LocalDate(yearMonthDayCalculator.GetYearMonthDay(days).WithCalendar(calendar));
+                LocalDate ret = new LocalDate(yearMonthDayCalculator.GetYearMonthDay(days).WithCalendar(calendar));
+                // The requested week year and the actual calendar year don't match, it could be due to short weeks
+                // in a non-regular rule. The simplest way to find out is just to check what the week year is...
+                // This problem doesn't apply to rules with regular weeks.
```

Comment:
Yes, given that we almost always use `!regular` that would make sense. Will update the comment first, then reverse the sense in a different commit.